File: 백준/Silver/10773. 제로/제로.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        int K = Integer.parseInt(bf.readLine());
        Stack<Integer> stack = new Stack<>();
        for(int i=0; i<K; i++) {
            int num = Integer.parseInt(bf.readLine());
            if(num == 0) {
                stack.pop();
            } else {
                stack.add(num);
            }
        }

        int result = stack.stream()
                            .mapToInt(i -> i)
                            .sum();

        System.out.println(result);

    }
}

File: 백준/Silver/10773. 제로/제로.py
import sys


def solution():
    stack = []
    K = int(sys.stdin.readline())
    for _ in range(K):
        num = int(sys.stdin.readline())
        if num == 0:
            stack.pop(-1)
        else:
            stack.append(num)
    print(sum(stack))
    return 0


if __name__ == '__main__':
    solution()